streams-contrib/streams-persist-cassandra/src/main/java/org/apache/streams/cassandra/CassandraClient.java

```diff
@@ -0,0 +1,160 @@
+import java.net.InetSocketAddress;
+import java.security.KeyStore;
+import java.util.Collection;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import static com.datastax.driver.core.SocketOptions.DEFAULT_CONNECT_TIMEOUT_MILLIS;
+import static com.datastax.driver.core.SocketOptions.DEFAULT_READ_TIMEOUT_MILLIS;
+
+public class CassandraClient {
+
+  private static final Logger LOGGER = LoggerFactory
+      .getLogger(CassandraClient.class);
+
+  private Cluster cluster;
+  private Session session;
+
+  public CassandraConfiguration config;
+
+  public CassandraClient(CassandraConfiguration config) throws Exception {
+    this.config = config;
+    org.apache.cassandra.config.Config.setClientMode(true);
+  }
+
+  public void start() throws Exception {
+
+    Preconditions.checkNotNull(config);
+
+    LOGGER.info("CassandraClient.start {}", config);
+
+    Cluster.Builder builder = Cluster.builder()
+        .withPort(config.getPort().intValue())
+        .withoutJMXReporting()
+        .withoutMetrics()
+        .withSocketOptions(
+            new SocketOptions()
+                .setConnectTimeoutMillis(DEFAULT_CONNECT_TIMEOUT_MILLIS*10)
+                .setReadTimeoutMillis(DEFAULT_READ_TIMEOUT_MILLIS*10)
+        );
+
+    if( config.getSsl() != null && config.getSsl().getEnabled() == true) {
+
+      Ssl ssl = config.getSsl();
+
+      KeyStore ks = KeyStore.getInstance("JKS");
+
+      InputStream trustStore = new FileInputStream(ssl.getTrustStore());
+      ks.load(trustStore, ssl.getTrustStorePassword().toCharArray());
+      InputStream keyStore = new FileInputStream(ssl.getKeyStore());
+      ks.load(keyStore, ssl.getKeyStorePassword().toCharArray());
+
+      TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+      tmf.init(ks);
+
+      KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+      kmf.init(ks, ssl.getKeyStorePassword().toCharArray());
+
+      SSLContext sslContext = SSLContext.getInstance("SSLv3");
+      sslContext.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+
+      SSLOptions sslOptions = JdkSSLOptions.builder()
+          .withSSLContext(sslContext)
+          .build();
+
+      builder = builder.withSSL(sslOptions);
+    }
+
+    Collection<InetSocketAddress> addresses = Lists.newArrayList();
+    for (String h : config.getHosts()) {
+      LOGGER.info("Adding Host: {}", h);
+      InetSocketAddress socketAddress = new InetSocketAddress(h, config.getPort().intValue());
+      addresses.add(socketAddress);
+    }
+    builder.addContactPointsWithPorts(addresses);
+
+    if( !Strings.isNullOrEmpty(config.getUser()) &&
+        !Strings.isNullOrEmpty(config.getPassword())) {
+      builder.withCredentials(config.getUser(), config.getPassword());
+    }
+    cluster = builder.build();
+
+    Preconditions.checkNotNull(cluster);
```

Comment:
i'll change it, to make it easier to remove once we have a substitute for Queues